weTranslate/SearchHeaderViewController.swift

```diff
@@ -67,24 +90,40 @@ final class SearchHeaderViewController: UIViewController {
         viewModel = SearchHeaderViewModel()
     }
 
+
+    // MARK: - UIResponder
+
+    override func resignFirstResponder() -> Bool {
+        state = .Default
+        return true
+    }
+
+
     // MARK: - Methods
 
     func swap(sender: UIButton?) {
         viewModel.swap()
+        search()
     }
 
     func selectLanguage(sender: UIButton?) {
-        toggle()
+        state = state == .Picker ? .Default : .Picker
     }
 
 
     // MARK: - Private
 
-    private func toggle() {
-        isExpended = !isExpended
-        self.searchHeaderView.languagesPickerView.alpha = self.isExpended ? 1 : 0
+    private func toggle(expended expended: Bool) {
+        self.searchHeaderView.languagesPickerView.alpha = expended ? 1 : 0
```

Comment:
Is there any way we can replace this inline condition by setting directly the value since bool and int express the same thing here.